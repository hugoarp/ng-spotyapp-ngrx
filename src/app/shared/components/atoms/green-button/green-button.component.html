<button class="green-button">
  <span class="green-button__label">{{ label }}</span>
</button>
