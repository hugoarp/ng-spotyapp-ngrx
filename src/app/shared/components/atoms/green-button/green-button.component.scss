@import "../../../../styles/variables.scss";

.green-button {
  background-color: $green-primary;
  width: 100%;
  padding: 0.5rem 0;
  font-weight: 600;
  font-size: 24px;
  border-radius: 2rem;
  &:hover {
    filter: brightness(1.1);
  }
}
