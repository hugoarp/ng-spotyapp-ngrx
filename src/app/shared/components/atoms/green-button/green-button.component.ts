import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-green-button',
  templateUrl: './green-button.component.html',
  styleUrls: ['./green-button.component.scss'],
})
export class GreenButtonComponent implements OnInit {
  @Input() label: string = '';

  constructor() {}

  ngOnInit(): void {}
}
