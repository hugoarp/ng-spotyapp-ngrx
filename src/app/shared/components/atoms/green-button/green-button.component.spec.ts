import { ComponentFixture, TestBed } from '@angular/core/testing';
import { GreenButtonComponent } from './green-button.component';

describe('GreenButtonComponent', () => {
  let component: GreenButtonComponent;
  let fixture: ComponentFixture<GreenButtonComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [GreenButtonComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(GreenButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('debe recibir un valor de tipo string en el input label', () => {
    const label = 'Hola Mundo';
    component.label = label;
    fixture.detectChanges();
    const button = compiled.querySelector('button');
    expect(button?.textContent).toBe(label);
  });
});
