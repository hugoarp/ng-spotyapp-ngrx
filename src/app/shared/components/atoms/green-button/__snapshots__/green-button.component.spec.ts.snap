// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenButtonComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <button
    class="green-button"
  >
    <span
      class="green-button__label"
    >
      
    </span>
  </button>
</div>
`;
